### ISSUE TITLE
Is the cache system actually working ?

### ISSUE THREAD

**Issue by hopye** [pgrimaud/instagram-user-feed#16]
Can you explain how the cache system works ? will it cache the data itself ? i would to avoid making a network connection to instagram whenever possible , in my case for another project i was downloading the data and storing into a cache file, then it would check if the data is older than lets say 4 hours, if so.. get the data remotely...

Great Scraper by the way! Thanks for taking the time!

**Comment by pgrimaud** [comment]
Hi @hopye,

The cache system stores data like **cookie** and **GIS (csrf token)** for paginate.

**TL;DR** : If you want to avoid making a network connection to instagram whenever possible. You have to manage cache of instagram data from your own side.

---

How works this library? With basic example :

```php
// Initialization

$cache = new Instagram\Storage\CacheManager('/path/to/your/cache/folder');
$api   = new Instagram\Api($cache);
$api->setUserName('pgrimaud');

// First call, it will scrap html page and extract a json file with the 12 first medias. 
// From this json file, it will save into cache system cookie, csrf token and user id.

$feed = $api->getFeed();

```

What is done in previous `$api->getFeed();` :

https://github.com/pgrimaud/instagram-user-feed/blob/master/src/Instagram/Transport/HtmlTransportFeed.php#L55

```php
// After html fetch call

$newCache = new Cache();
$newCache->setRhxGis($data->rhx_gis);
$newCache->setCookie($res->getHeaders()['Set-Cookie']);
$newCache->setUserId($data->entry_data->ProfilePage[0]->graphql->user->id);

```

Cache is used when you use paginate : 

```php
// EndCursor is used for paginate only

$endCursor = $feed->getEndCursor();
$api->setEndCursor($endCursor);

// Second call, it will fetch data from cache to generate json route
// Then fetch it
$feed = $api->getFeed();

```

What is done in previous `$api->getFeed();` :

https://github.com/pgrimaud/instagram-user-feed/blob/master/src/Instagram/Transport/JsonTransportFeed.php#L50

```php
// Before json fetch call

$variables = [
    'id'    => $cache->getUserId(),
    'first' => '12',
    'after' => $this->endCursor,
];
$cookieJar = CookieJar::fromArray($cache->getCookie(), 'www.instagram.com');

$headers = [
    'headers' => [
        'user-agent'       => self::USER_AGENT,
        'x-requested-with' => 'XMLHttpRequest',
        'x-instagram-gis'  => $this->generateGis($cache->getRhxGis(), $variables)
    ],
    'cookies' => $cookieJar
];

```

---

**Conclusion : If you want to avoid making a network connection to instagram whenever possible. You have to manage cache of instagram data from your own side.**

Hope it helps 😄
